tests/testthat/test-rectangle.R

```diff
@@ -147,6 +147,14 @@ test_that("list of 0-length vectors yields no new columns", {
   expect_named(unnest_wider(df, y), "x")
 })
 
+test_that("list_of columns can be unnested", {
+  df <- tibble(x = 1:2, y = list_of(1L, 1:2))
```

Comment:
Maybe make this `list_of(c(a = 1L), c(a = 1L, b = 2L))` to get better names in the output?